### PR TITLE
feat: lock retry

### PR DESCRIPTION
**Pull request by varas**
It adds `lock_retries` with *default* value being `30` when no arg is provided, still `0` is accepted to fail on error.
It uses a fixed retry backoff of 1min.

---

It also paramtrizes AWS lock resource tags via `aws_tags` by default same ones as before.

### CODE REVIEW

File: publisher/lock/s3_lock.go [hunk 1]
```diff
@@ -55,37 +54,60 @@ func (l *lockData) isExpired(ttl time.Duration, t time.Time) bool {
...
 func (l *S3) Lock() error {
+	var tries uint
+retry:
+	if (l.conf.MaxRetries - tries) > 0 {
+		tries++
+		l.logF("%s attempt %d", l.conf.Owner, tries)
+		if l.isBusyDeletingExpired() {
+			l.logF("%s failed, waiting %s", l.conf.Owner, l.conf.RetryBackoff.String())
+			time.Sleep(l.conf.RetryBackoff)
+			goto retry
```
Comment (gsanchezgavier) [inline publisher/lock/s3_lock.go:101]: I would replace the `goto` for a `for`+break loop iterating over the MaxRetries just to be a more easy to read. btw i didn't know that `goto` exists on Go. Just a suggestion.

**Review by gsanchezgavier — COMMENTED** [review]
Thanks for work on the retry mechanism! 
Could you add the parameters to the action.yml ?

**Comment by gsanchezgavier** [comment]
@varas  could we keep the same parameter name `disable_lock` and make it always retry on busy? We can add a new parameter `max_retry` with an x number of retries (that if you set it to 0 would be the same as `fail_on_busy`)
Doing this we do not add a breaking change, we keep the v1 and all the integration pipelines will going to get this without any changes.
wdyt?

**Comment by varas** [comment]
@gsanchezgavier breaking change removed at https://github.com/newrelic/infrastructure-publish-action/pull/84

File: publisher/lock/s3_lock.go (hunk 1 above)
Comment (varas) [reply publisher/lock/s3_lock.go:101]: https://github.com/newrelic/infrastructure-publish-action/pull/85

**Review by gsanchezgavier — APPROVED** [review]
Thanks for adding this feature!

**Pull request merged**